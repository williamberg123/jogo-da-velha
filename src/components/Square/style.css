.Square {
    background-color: rgb(116, 212, 116);
    display: flex;
    align-items: center;
    justify-content: center;
    aspect-ratio: 1/1;
}

.Square img {
    width: 80%;
}