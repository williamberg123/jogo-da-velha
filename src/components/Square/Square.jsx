import React from 'react';

import PropTypes from 'prop-types';

import './style.css';

export default function Square({ funcHandleClick, index }) {
    return (
        <div data-testid="Square" onClick={funcHandleClick} className={`Square Square${index}`}>
        </div>
    );
}

Square.propTypes = {
    funcHandleClick: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired
};
