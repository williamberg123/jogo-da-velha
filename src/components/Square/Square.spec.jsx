import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import Square from './Square';

describe('<Square />', () => {
    it('should call function when div is clicked', () => {
        const fn = jest.fn();
        render(<Square funcHandleClick={fn} index={1} />);

        fireEvent.click(screen.getByTestId('Square'));

        expect(fn).toBeCalled();
    });

    it('should to have the class Square1', () => {
        const fn = jest.fn();
        render(<Square funcHandleClick={fn} index={1} />);

        expect(screen.getByTestId('Square')).toHaveClass('Square1');
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        render(<Square funcHandleClick={fn} index={1} />);

        expect(screen.getByTestId('Square')).toMatchSnapshot();
    });
});
