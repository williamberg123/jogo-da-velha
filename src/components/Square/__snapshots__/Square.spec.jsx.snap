// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> should match snapshot 1`] = `
<div
  class="Square Square1"
  data-testid="Square"
/>
`;
